pkg/updater/download.go

```diff
@@ -73,41 +104,87 @@ func (d *downloader) Download(ctx context.Context, pkg Package, destinationPath
 	if err != nil {
 		return fmt.Errorf("could not write archive file: %w", err)
 	}
+
+	// Verify hash
 	sha256 := hashWriter.Sum(nil)
 	expectedHash, err := hex.DecodeString(pkg.SHA256)
 	if err != nil {
 		return fmt.Errorf("could not decode hash: %w", err)
 	}
 	if !bytes.Equal(expectedHash, sha256) {
-		return fmt.Errorf("invalid hash for %s: expected %x, got %x", pkg.URL, pkg.SHA256, sha256)
+		return fmt.Errorf("invalid hash for %s: expected %s, got %x", pkg.URL, pkg.SHA256, sha256)
+	}
+
+	// Extract OCI archive to a temporary directory
+	extractedArchivePath := filepath.Join(tmpDir, "oci")
+	if err := os.Mkdir(extractedArchivePath, 0755); err != nil {
+		return fmt.Errorf("could not create archive extraction directory: %w", err)
 	}
-	err = extractTarGz(archivePath, destinationPath)
+
+	err = extractTarXz(archivePath, extractedArchivePath)
 	if err != nil {
 		return fmt.Errorf("could not extract archive: %w", err)
 	}
+
+	// Extract package from OCI archive
+	extractedOCIPath := filepath.Join(tmpDir, "extracted")
+	if err := os.Mkdir(extractedOCIPath, 0755); err != nil {
+		return fmt.Errorf("could not create OCI extraction directory: %w", err)
+	}
+
+	err = extractOCI(extractedArchivePath, extractedOCIPath)
```

Comment:
Couldn't storage usage be reduced a bit here by removing the orignal compressed archive before calling this function ?

Reply:
I'll discuss with agent folks generating the archive, but this is achievable